filebeat/beater/filebeat.go

```diff
@@ -82,19 +81,27 @@ func (fb *Filebeat) Run(b *beat.Beat) error {
 		logp.Err("Could not start registrar: %v", err)
 	}
 	// Stopping registrar will write last state
-	defer registrar.Stop()
+	defer func() {
```

Comment:
@urso can you have a really close look at the following defer statements which I have changed? I moved the `registrarLogger.Close()`which was `sucessLogger `before, before the registrar Stopping as I think that makes more sense. Same change I did for the publisherChan below.